metrics/prometheus.go

```diff
@@ -1156,7 +1156,17 @@ func (c *PrometheusCollector) collectContainersInfo(ch chan<- prometheus.Metric)
 		labels := make([]string, 0, len(rawLabels))
 		containerLabels := c.containerLabelsFunc(cont)
 		for l := range rawLabels {
-			labels = append(labels, sanitizeLabelName(l))
+			duplicate := false
+			sl := sanitizeLabelName(l)
+			for _, x := range labels {
+				if sl == x {
+					duplicate = true
+					break
+				}
+			}
+			if !duplicate {
+				labels = append(labels, sl)
```

Comment:
can we end up with fewer labels than values?  Should we move the `values = append(values, containerLabels[l])` statement inside here as well?

Reply:
I was wondering that after the test failure, but this shouldn't change that behavior right?  It will only exclude something if it a label if one already exists and the value will still get set for that label.

Reply:
I would be surprised if prometheus didn't yell at us if we tried to, for example, use a description with 3 labels, but then provide 4 label values when creating the metric.  What you are implicitly doing here is just using the first occurrence of a given sanitized label, and ignoring subsequent ones.

Reply:
Hmm, yeah I see. It could happen if someone provides multiple permutations of a label that all normalize to the same thing. Should we just throw them out in that case? I can't think of a great default behavior there.

Reply:
^ on the same container, it should hopefully be a rare edge-case...

Reply:
wasn't that exactly what your bug was?

Reply:
I think just picking the first is a fine behavior.

Reply:
No #2181 was node-wide. 2 separate containers (separate pods) with annotations that normalize to the same thing caused the panic.

Reply:
Ok, I get the difference now.  I think we would still always get more values than labels, since if the label isn't present, we still add it with the value "".

Reply:
That's true, in fact I noticed when running without `--store_container_labels=false` all labels that were present on _any_ of the containers on the host showed up with empty values for all container metrics in prometheus, that was what made me look into the cgroup whitelisting initially and then I noticed this crash behavior.

Reply:
Yeah... Prometheus requires that all metric streams in a given scrape have the same set of labels.  So our workaround is just to add empty values for all labels we don't have.